### PR TITLE
fix: support running multiple verify tests with same image

### LINKED ISSUE #7640: "skaffold verify" erroring when same container (image+tag) is reused in multiple steps (eg: alpine:latest)
Currently if "skaffold verify" is run with the same container-image+tag (eg: alpine:3 in the below example) skaffold verify will have some issues running-containers/reading-logs as there is mutliple containers with duplicate names.  

skaffold.yaml `verify` snippet:
```
verify:
- name: integration-test
  container:
    name: integration-test-container
    image: alpine:3
    command: ["/bin/false"]
- name: environment-variables
  container:
    name: environment-variables-container
    image: alpine:3
    command: ["/bin/echo"]
    args: ["$GOOGLE_APPLICATION_CREDENTIALS"]
- name: random-print
  container:
    name: randomly-printing
    image: alpine:3
    command: ["/bin/echo"]
    args: ["calling it a day!"]
```

shell error:
```
$ skaffold verify
Tags used in verification:
3: Pulling from library/alpine
Digest: sha256:686d8c9dfa6f3ccfc8230bc3178d23f84eeaf7e457f36f271ab1acc53015037c
Status: Image is up to date for alpine:3
3: Pulling from library/alpine
Digest: sha256:686d8c9dfa6f3ccfc8230bc3178d23f84eeaf7e457f36f271ab1acc53015037c
Status: Image is up to date for alpine:3
3: Pulling from library/alpine
Digest: sha256:686d8c9dfa6f3ccfc8230bc3178d23f84eeaf7e457f36f271ab1acc53015037c
Status: Image is up to date for alpine:3
[alpine:3] calling it a day!
2 error(s) occurred:
* creating container in local docker: Error response from daemon: Conflict. The container name "/alpine-6" is already in use by container "6d2c78907132b4775089dd1a87742d608174982ab9a049a27726fc9e15ce8cd4". You have to remove (or rename) that container to be able to reuse that name.
* verify test failed: "integration-test" running container image "alpine:3" errored during run with status code: 1
```

The best fix here is likely to modify the container name to instead be the `verify-test-name` although I as verify is built on the docker deployer this will likely be nontrivial

### PR DESCRIPTION
**Pull request by aaron-prindle**
Fixes #7640

### CODE REVIEW

**Review by MarlonGamez — COMMENTED** [review]
looks good, just have a couple qs

File: pkg/skaffold/verify/docker/verify.go [hunk 1]
```diff
@@ -257,7 +267,7 @@ func (v *Verifier) getContainerName(ctx context.Context, name string) string {
 		if !v.client.ContainerExists(ctx, currentName) {
 			break
 		}
-		currentName = fmt.Sprintf("%s-%d", name, counter)
+		currentName = fmt.Sprintf("%s-%s", name, uuid.New().String()[0:8])
 		counter++
```
[the remark this answers is not in the document]
Comment (aaron-prindle) [reply pkg/skaffold/verify/docker/verify.go]: sorry, I meant to get rid of this but it must have got lost in the shuffling.  Thanks for catching this, removed the `counter` var now

File: pkg/skaffold/verify/docker/verify.go [hunk 2]
```diff
@@ -257,7 +267,7 @@ func (v *Verifier) getContainerName(ctx context.Context, name string) string {
 		if !v.client.ContainerExists(ctx, currentName) {
 			break
 		}
-		currentName = fmt.Sprintf("%s-%d", name, counter)
+		currentName = fmt.Sprintf("%s-%s", name, uuid.New().String()[0:8])
```
[the remark this answers is not in the document]
Comment (aaron-prindle) [reply pkg/skaffold/verify/docker/verify.go:269]: I could use the whole uuid which is 32 chars - [RFC 4122](https://datatracker.ietf.org/doc/html/rfc4122) but I figured that using the first 8 bytes gives us enough uniqueness, allows the containerNames to be closer to the 128 char names docker allows (we now allow 120 chars) and is easier to type/view:
`alpine-f81d4fae` vs
`alpine-f81d4fae-7dec-11d0-a765-00a0c91e6bf6`

[the remark this answers is not in the document]
Comment (MarlonGamez) [reply pkg/skaffold/verify/docker/verify.go:269]: sounds good! Maybe just leave a comment as it's not immediately clear. Not necessary tho if you don't feel like it

**Review by MarlonGamez — APPROVED** [review]
lgtm

**Pull request merged**